scheduler/src/cook/mesos/task.clj

```diff
@@ -52,17 +54,18 @@
    2. The Cook executor command has been configured,
    3. Either :job/executor is explicitly enabled
       Or: the job is a cook-executor candidate (see cook-executor-candidate?)."
-  [job-ent {:keys [command portion retry-limit]}]
+  [job-ent]
   (and (not (use-custom-executor? job-ent))
-       command
+       (-> config :settings :executor :command)
```

Comment:
How do you feel about `def-state`-ing `executor-config`? Then  we do not need to navigate the config to retrieve the command.

Reply:
Should every sub-element in the config be its own `defstate`?

Reply:
Every sub-component could be, that let's us track dependencies better rather and also avoids passing the responsibility to the consumer to navigate the settings and retrieving the element it needs.

Reply:
Discussed offline, instead of `defstate`ing the executor config, I added a new function, `cook.config/executor-config`, which returns the executor config from the full config, and can be `with-redef`'d